clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtils.java

```diff
@@ -203,20 +201,19 @@ public static byte[] compress(byte[] raw, CompressionType compressionType) throw
 
     public static ByteBuffer decompress(byte[] metrics, CompressionType compressionType) {
         ByteBuffer data = ByteBuffer.wrap(metrics);
-        try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create());
-            ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-
+        try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create())) {
             byte[] bytes = new byte[data.capacity() * 2];
             int nRead;
             while ((nRead = in.read(bytes, 0, bytes.length)) != -1) {
-                out.write(bytes, 0, nRead);
+                try (ByteBufferOutputStream out = new ByteBufferOutputStream(nRead)) {
```

Comment:
Could we use single try-catch to release both `in` and `out`?

Reply:
The helper function already comes with [unit tests](https://github.com/apache/kafka/blob/bf9a27fefdb3d93c7a510f871433c4c9e07de71a/clients/src/test/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtilsTest.java#L126).